packages/studio/src/schemas/page.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod";
+
+export const getEditPageSchema = z.object({
+  pageId: z.string(),
+  siteId: z.string()
+})
+
+export const updatePageSchema = z.object({
+  // NOTE: We allow both to be empty now, 
+  // in which case this is a no-op. 
+  // We are ok w/ this because it doesn't 
+  // incur any db writes
```

Comment:
wait i am confused here. isnt every call to the upstream api call `updatePage` updating the db? in what situation will these two be nullable ah?

Reply:
if your parent id is null, this implies you're a root level page. 

if your `pageName` is null, this implies you're not renaming a page. if this is confusing, i'm ok w/ making it non nullable but that means that the caller always has to pass in these properties. 

the principle for this API call becomes "pass in what you want to update" to "pass in the current state of the world". 

i think both are fine, no point optimising for request size because it's never going to be big but i thought the former was clearer.

Reply:
Are we talking about something like a PATCH vs PUT?

Reply:
Also when I update a page schema, why do I need to pass in the parentId? From earlier discussion, it seems that we will derive the parentId from the page's resourceId